src/hooks/useRescanStatus.ts

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useRef, useState } from 'react'
+import { getrescaninfo } from '@/lib/jm-api/generated/client/sdk.gen'
+import { setSession, type RescanSession } from '@/lib/session'
+import { useApiClient } from './useApiClient'
+import { useSession } from './useSession'
+
+const RESCAN_PROGRESS_INTERVAL = 2000
+
+export const useRescanStatus = () => {
+  const session = useSession()
+  const client = useApiClient()
+  const completionTimeoutRef = useRef<NodeJS.Timeout | null>(null)
+  const [showCompletionMessage, setShowCompletionMessage] = useState(false)
```

Comment:
```suggestion
  const [showCompletionMessage, setShowCompletionMessage] = useState(false)
```

This is of no concern to this hook. It does not care what other components are doing. It should just provide the rescan data, no? What do you think?

Reply:
Yes if we remove the completion message then this entire logic wont be needed to begin with